cps/src/main/java/org/jenkinsci/plugins/workflow/cps/replay/ReplayAction.java

```diff
@@ -0,0 +1,242 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.workflow.cps.replay;
+
+import com.cloudbees.diff.Diff;
+import com.google.common.collect.ImmutableList;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import hudson.Extension;
+import hudson.Functions;
+import hudson.init.InitMilestone;
+import hudson.init.Initializer;
+import hudson.model.Action;
+import hudson.model.Cause;
+import hudson.model.CauseAction;
+import hudson.model.Item;
+import hudson.model.Job;
+import hudson.model.ParametersAction;
+import hudson.model.Run;
+import hudson.model.queue.QueueTaskFuture;
+import hudson.security.Permission;
+import hudson.security.PermissionScope;
+import hudson.util.FormValidation;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.CheckForNull;
+import javax.servlet.ServletException;
+import jenkins.model.Jenkins;
+import jenkins.model.ParameterizedJobMixIn;
+import jenkins.model.TransientActionFactory;
+import jenkins.scm.api.SCMRevisionAction;
+import net.sf.json.JSON;
+import net.sf.json.JSONObject;
+import org.acegisecurity.AccessDeniedException;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowExecution;
+import org.jenkinsci.plugins.workflow.cps.replay.Messages;
+import org.jenkinsci.plugins.workflow.flow.FlowExecution;
+import org.jenkinsci.plugins.workflow.flow.FlowExecutionOwner;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+/**
+ * Attached to a {@link Run} when it could be replayed with script edits.
+ */
+@SuppressWarnings("rawtypes") // on Run
+public class ReplayAction implements Action {
+
+    private final Run run;
+
+    private ReplayAction(Run run) {
+        this.run = run;
+    }
+
+    @Override public String getDisplayName() {
+        return "Replay";
+    }
+
+    @Override public String getIconFileName() {
+        return isEnabled() ? "clock.png" : null;
```

Comment:
Good idea. There is no 32×32 but maybe it does not matter. Will check how it looks.